src/mon/OSDMonitor.cc

```diff
@@ -1127,6 +1127,8 @@ void OSDMonitor::on_shutdown()
   list<MonOpRequestRef> ls;
   take_all_failures(ls);
   ls.clear();
+
+  cct->_conf.remove_observer(this);
```

Comment:
`add_observer()` was included in #28227.  and we call it in the constructor. i think the correct fix is to call it in the destructor. also, i cannot find any rationale of this change. for instance, is this a cleanup or a fix? if it is a fix, what the impact of the issue? shall we backport it ?

Reply:
> `add_observer()` was included in #28227. and we call it in the constructor. i think the correct fix is to call it in the destructor. 

As discussed with Sridhar, it is unsure why OSDMonitor doesn't override `PaxosService::init()`, which causes this. I think it was there but dropped later.

> also, i cannot find any rationale of this change. for instance, is this a cleanup or a fix? if it is a fix, what the impact of the issue? shall we backport it ?

This is just a clean up, found while working on something else.